### PR TITLE
Pull latest NTRU Prime from supercop-20210529

### PR DESCRIPTION
**Pull request by jschanck**
Integrate the latest NTRU Prime code. Upstream has made some performance improvements, but the code is interoperable with previous releases and does not add any parameter sets (although other parameter sets are available upstream).

I've removed an unnecessary AES shim that my packaging script was writing to the sntrup directories.

Known issues: clang-tidy doesn't like the use of recursion in Fq_pow and int32_sort. I can add NOLINT annotations, but I'd actually prefer to disable the warning.

### CODE REVIEW

File: crypto_kem/ntrulpr761/clean/kem.c [hunk 1]
```diff
@@ -156,7 +158,9 @@ static void Hide(unsigned char *c, unsigned char *r_enc, const Inputs r, const u
             s[0] = 5;
             Hash(h, s, sizeof s);
         }
-        PQCLEAN_NTRULPR761_CLEAN_crypto_stream_aes256ctr((unsigned char *) L, 4 * p, aes_nonce, h);
+        if (PQCLEAN_NTRULPR761_CLEAN_crypto_stream_aes256ctr((unsigned char *) L, 4 * p, aes_nonce, h) != 0) {
+            abort();
+        }
```
Comment (thomwiggers) [inline crypto_kem/ntrulpr761/clean/kem.c:163]: could this `abort()` not better be handled in the function that wraps `aes256ctr`?

**Review by thomwiggers — COMMENTED** [review]
Thanks for the update.

I've got one small question.

File: crypto_kem/ntrulpr761/clean/kem.c (hunk 1 above)
Comment (jschanck) [reply crypto_kem/ntrulpr761/clean/kem.c]: Thanks. Upstream might call OpenSSL's aes directly at that line, but the abort isn't necessary with the wrapper. I've removed it.

**Review by thomwiggers — APPROVED** [review]

**Comment by dstebila** [comment]
This code has now propagated to the OQS benchmarking and @baentsch has observed an improvement in performance for NTRU Prime decaps but a slow down in performance for other aspects of NTRU Prime.  Is that expected?

**Comment by baentsch** [comment]
Here [a link to the performance numbers before](https://openquantumsafe.org/benchmarking/visualization/2021-07-14/speed_kem.json) and [here a link to new numbers after this new code got added](https://openquantumsafe.org/benchmarking/visualization/2021-07-15/speed_kem.json).

**Comment by thomwiggers** [comment]
Those JSON dumps weren't very accessible, so here I've at least pretty-printed them: https://gist.github.com/thomwiggers/a85c372fdef9d272614aca20f1993a79

**Pull request merged**